lib/tls.c

```diff
@@ -528,9 +548,23 @@ int rc_init_tls(rc_handle * rh, unsigned flags)
 	SERVER *authservers;
 	char hostname[256];	/* server's hostname */
 	unsigned port;		/* server's port */
+#ifdef __linux__    
+	char *ns = NULL;
+	int ns_def_hdl = 0;
+#endif    
 
 	memset(&rh->so, 0, sizeof(rh->so));
 
+#ifdef __linux__    
+	ns = rc_conf_str(rh, "namespace"); /* Check for namespace config */
```

Comment:
What if you allow `rc_conf_str(namespace)` and `rc_set_netns()` unconditionally everywhere and have the `#ifdef __linux__` block inside `rc_set_netns` (and return error otherwise)? It may reduce the number of ifdefs which are not spread through the code.